test/index.ts

```diff
@@ -836,8 +837,20 @@ function testNonEmptyObject() {
   type ObjectWithKeys = { foo: string; bar: number; xyz: undefined };
   type EmptyObject = {};
 
-  type TestA = Assert<IsExact<NonEmptyObject<ObjectWithKeys>, ObjectWithKeys>>;
-  type TestB = Assert<IsExact<NonEmptyObject<EmptyObject>, never>>;
+  type Cases = [
```

Comment:
If you have time, as you've added more tests, could you please move them to a separate file, e.g. `non-empty-object.ts`?

If not, feel free to resolve this comment!

Reply:
Sure will refactor and move it to separate test file!